### ISSUE TITLE
`testing-library/no-await-sync-events` reporting false positives for `@testing-library/user-event` v14

### ISSUE THREAD

**Issue by swissspidy** [testing-library/eslint-plugin-testing-library#567]
### Plugin version

5.0.5

### ESLint version

8.12.0

### Node.js version

16.4.2

### npm/yarn version

8.5.0

### Operating system

macOS Monterey, 12.3

### Bug description

In version 14, `@testing-library/user-event` changed all APIs to always return a Promise.

See https://github.com/testing-library/user-event/releases/tag/v14.0.0

### Steps to reproduce

Use the async APIs in `@testing-library/user-event` v14 and see all the false positives reported by this rule

### Error output/screenshots

```
error  `userEvent.click` is sync and does not need `await` operator  testing-library/no-await-sync-events
```

### ESLint configuration

{
  "testing-library/no-await-sync-events": "error",
}

### Rule(s) affected

testing-library/no-await-sync-events

### Do you want to submit a pull request to fix this bug?

No

**Comment by Belco90** [comment]
Thanks for reporting @swissspidy. I've been thinking about this since I saw the beta of the new `user-event`.

Ideally, this plugin should guess which `user-event` you are using to report its methods or not in this rule.

However, since we don't have that mechanism in place, I think including a new option to this rule would work as quick workaround. We can add a new option called `eventModules` accepting an array of string, which defaults to `["fire", "user"]`, so it can be customized to skip reporting `user-event`.

The naming has to be improved, bit I think I'll go for this workaround.

**Comment by MichaelDeBoey** [comment]
@Belco90 I think we should add a setting to say which version of `user-event` is used and try to auto-detect it, like we have the `react` version number

**Comment by Belco90** [comment]
@MichaelDeBoey sure, but that requires more time. I can implement the option mentioned in this rule fairly quick, then we can include the mechanism set/auto-detect the `user-event` version and toggle the things to be reported based on it.

**Comment by weyert** [comment]
Sorry, what's the expected configuration to make `await userEvent.type()` work? I tried the following configuration but it still throwing the error:

```json
        'testing-library/no-await-sync-events': [
          "error",
          {
            "eventModules": ["fire-event", "user-event"]
          }
        ],
```

**Comment by Belco90** [comment]
@weyert [As indicated in the rule docs](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-await-sync-events.md#notes), you need to set `eventModules` to just `["fire-event"]`.
